### ISSUE TITLE
SDML does not work in sandwich example

### ISSUE THREAD

**Issue by bellet** [metric-learn/metric-learn#138]
#### Description
SDML does not seem to do anything in the sandwich example from the doc (see figure below). We should make sure it is due to the default hyperparameters (and fix them in the example is that's the case).

![sphx_glr_plot_sandwich_001](https://user-images.githubusercontent.com/7905665/50301250-1de2a700-0487-11e9-9525-4a6cd9706fd1.png)

**Comment by wdevazelhes** [comment]
In a previous version of metric-learn, scipy, and numpy, the example seemed to work. For instance, I have this result for theses versions: 
Linux-4.4.0-141-generic-x86_64-with-debian-stretch-sid
('Python', '2.7.15 |Anaconda, Inc.| (default, Dec 14 2018, 19:04:19) \n[GCC 7.3.0]')
('NumPy', '1.11.3')
('SciPy', '0.17.1')
('Scikit-Learn', '0.17.1')
metric-learn: from commit ddfac991c0d32f3931831d649133dfbd15c61eff

![figure_1-2](https://user-images.githubusercontent.com/31916524/51022270-1729ea00-1584-11e9-9bf5-fcc4ec62159a.png)

I guess there is some problem coming from new versions of either metric-learn, scikit-learn, scipy or numpy...

**Comment by wdevazelhes** [comment]
Related to #154
